genson/src/main/java/com/owlike/genson/ext/jackson/JacksonBundle.java

```diff
@@ -18,61 +28,177 @@
+                return Trilean.TRUE;
+            }
+            return Trilean.UNKNOWN;
         }
 
         @Override
         public Trilean isAccessor(Field field, Class<?> fromClass) {
-            return null;
+            if (ignore(field, field.getType())) {
+                return Trilean.FALSE;
+            }
+            if (include(field, field.getType())) {
+                return Trilean.TRUE;
+            }
+            return Trilean.FALSE;
         }
 
         @Override
         public Trilean isAccessor(Method method, Class<?> fromClass) {
-            return null;
+            if (ignore(method, method.getReturnType())) return Trilean.FALSE;
+
+            String name = null;
+            if (method.getName().startsWith("get") && method.getName().length() > 3)
+                name = method.getName().substring(3);
+            else if (method.getName().startsWith("is") && method.getName().length() > 2
+                    && method.getReturnType() == boolean.class
+                    || method.getReturnType() == Boolean.class)
+                name = method.getName().substring(2);
+
+            if (name != null) {
+                if (include(method, method.getReturnType())) return Trilean.TRUE;
+                if (find(JsonIgnore.class, fromClass, "set" + name, method.getReturnType()) != null)
+                    return Trilean.FALSE;
+            }
+
+            return Trilean.FALSE;
         }
 
         @Override
         public Trilean isMutator(Field field, Class<?> fromClass) {
-            return null;
+            if (ignore(field, field.getType())) {
+                return Trilean.FALSE;
+            }
+            if (include(field, field.getType())) {
+                return Trilean.TRUE;
+            }
+            return Trilean.FALSE;
         }
 
         @Override
         public Trilean isMutator(Method method, Class<?> fromClass) {
+            Class<?> paramClass = method.getParameterTypes().length == 1 ? method
+                    .getParameterTypes()[0] : Object.class;
+            if (ignore(method, paramClass)) return Trilean.FALSE;
+
+            if (method.getName().startsWith("set") && method.getName().length() > 3) {
+                if (include(method, method.getReturnType())) return Trilean.TRUE;
+
+                String name = method.getName().substring(3);
+
+                // Exclude it if there is a corresponding accessor annotated with JsonIgnore
+                if (find(JsonIgnore.class, fromClass, "get" + name) != null) return Trilean.FALSE;
+                if (paramClass.equals(boolean.class) || paramClass.equals(Boolean.class)) {
+                    if (find(JsonIgnore.class, fromClass, "is" + name) != null)
+                        return Trilean.FALSE;
+                }
+            }
+
+            return Trilean.FALSE;
+        }
+
+        // ---- private methods ---------------------------------------------
```

Comment:
These methods seem to be reusable across various implementations that rely on annotations, and may be somewhat duplicated already.

Can we pull them into a utility class or better yet, an interface that all resolvers can then implement and reuse, so we can refactor the existing ones and remove some of the code duplication?

For example:
```java
public interface JsonAnnotations {
    <A extends Annotation> A includeAnnotation();
    <A extends Annotation> A excludeAnnotation();

    default boolean ignore(AccessibleObject property, Class<?> ofType) {
        return find(excludeAnnotation(), property, ofType) != null;
    }

    default boolean include(AccessibleObject property, Class<?> ofType) {
        return find(includeAnnotation(), property, ofType) != null;
    }
 
    static <A extends Annotation> A find(Class<A> annotation, AccessibleObject onObject, Class<?> onClass) {
        A ann = onObject.getAnnotation(annotation);
        if (ann != null) return ann;
        return find(annotation, onClass);
    }

    // etc...
}
```
Then various resolvers could simply implement the interface above and implement the methods that return annotation classes that should be used for inclusion and exclusion.

Thoughts?